src/main/java/com/iexec/worker/result/ResultService.java

```diff
@@ -443,4 +429,49 @@ public String computeResultDigest(ComputedFile computedFile) {
         }
         return resultDigest;
     }
+
+    // region Purge
+
+    /**
+     * Purge results from given task, especially its result folder.
+     *
+     * @param chainTaskId ID of the task to purge.
+     * @return {@literal true} if task folder has been deleted
+     * and task data has been cleared from this service;
+     * {@literal false} otherwise.
+     */
+    @Override
+    public boolean purgeTask(String chainTaskId) {
+        final String taskBaseDir = workerConfigService.getTaskBaseDir(chainTaskId);
+
+        resultInfoMap.remove(chainTaskId);
+        FileHelper.deleteFolder(taskBaseDir);
+
+        final boolean deletedInMap = !resultInfoMap.containsKey(chainTaskId);
+        final boolean deletedTaskFolder = !new File(taskBaseDir).exists();
+
+        boolean deleted = deletedInMap && deletedTaskFolder;
+        if (deletedTaskFolder) {
+            log.info("The result of the chainTaskId has been deleted [chainTaskId:{}]", chainTaskId);
+        } else {
+            log.warn("The result of the chainTaskId couldn't be deleted [chainTaskId:{}, deletedInMap:{}, " +
+                            "deletedTaskFolder:{}]",
+                    chainTaskId, deletedInMap, deletedTaskFolder);
+        }
+
+        return deleted;
+    }
+
+    /**
+     * Purge results from all known tasks, especially their result folders;
+     */
+    @Override
+    public void purgeAllTasksData() {
```

Comment:
When is this method used? When the worker restarts?

Reply:
Basically, when the worker stops. So this includes the case when the worker restarts, yes. Please see [`Purgeable` interface](https://github.com/iExecBlockchainComputing/iexec-common/blob/5ce907e180d3d28b2acf2d3e3a584c3f8d14ff07/src/main/java/com/iexec/common/lifecycle/purge/Purgeable.java#L19).